plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/PluginWithIntegrationTestsHook.java

```diff
@@ -1,24 +1,40 @@
 package org.jenkins.tools.test.hook;
 
-import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
-
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
 /**
- * Workaround for those plugins with integration tests since they need execute the failsafe:integration-test goal before execution.
+ * Hook for plugins with integration tests that need to be executed 
  */
 public abstract class PluginWithIntegrationTestsHook extends PluginCompatTesterHookBeforeExecution {
 
+    /** Inform about goals to execute integration tests */
+    @NonNull
+    abstract public Collection<String> getGoals();
+    
+    /** Inform about test type suite to execute integration test */
+    @NonNull
+    abstract public Collection<String> getTestTypes();
+    
+    @SuppressWarnings("unchecked")
     @Override
     public Map<String, Object> action(Map<String, Object> info) throws Exception {
         List<String> args = (List<String>) info.get("args");
 
         if (args != null) {
-            args.add("failsafe:integration-test");
+            args.addAll(getGoals());
+        }
+        
+        List<String> types = (List<String>) info.get("types");
+        if (types != null) {
+            types.addAll(getTestTypes());
         }
-
         return info;
     }
-
+    
```

Comment:
Extra spaces 😄

Reply:
Thanks! fixed in https://github.com/jenkinsci/plugin-compat-tester/pull/258/commits/a917eb1355b4e32abdf0db565b666b3a43c2d23d